src/tests/ftest/rebuild/basic.py

```diff
@@ -23,12 +22,12 @@ def run_rebuild_test(self, pool_quantity):
             pool_quantity (int): number of pools to test
         """
         # Get the test parameters
-        self.pool = []
+        pools = []
```

Comment:
Why do we use `pools = []` instead of `self.pool = []`? We're using `self.container = []` for container.

Reply:
This is part of a larger effort to cleanup the way pools and containers are managed.
1. Some tests use a single `self.pool = <TestPool>` and some use multiple `self.pool = [<TestPool>, ...]`, which makes common utility files confusing.
2. There isn't a real need to store into `self.pool`. It's better to pass a `pool` argument to functions as needed for readability, modularity, maintenance, etc. It's just not ideal design to store everything in `self.`
3. `self.container` will eventually be cleaned up similarly. But none of this is formalized yet

Reply:
I understand it's confusing if we store list into `self.pool`, but if it's a single pool, don't we want to use `self.pool` because the pool assigned to it will be cleaned up during tearDown?

Reply:
I see. Thanks!